File: libs/monobitmap/src/main/kotlin/mono/graphics/bitmap/drawable/Drawable.kt
package mono.graphics.bitmap.drawable

import mono.graphics.bitmap.MonoBitmap

/**
 * An interface for drawable which is the minimal version of a bitmap. A drawable contains enough
 * information for generating any-size bitmap.
 */
interface Drawable {
    fun toBitmap(width: Int, height: Int): MonoBitmap
}

File: monosketch-svelte/src/lib/mono/monobitmap/drawable/drawable.ts
import { MonoBitmap } from "$mono/monobitmap/bitmap/monobitmap";

/**
 * An interface for drawable which is the minimal version of a bitmap. A drawable contains enough
 * information for generating any-size bitmap.
 */
export interface Drawable {
    toBitmap(width: number, height: number): MonoBitmap.Bitmap;
}
